apps/condo/domains/common/components/tasks/TaskProgress.tsx

```diff
@@ -141,12 +141,23 @@ export const TaskProgressTracker: React.FC<ITaskProgressTrackerProps> = ({ task
         }
     })
 
-    const handleRemoveTask = useCallback(() => {
+    const removeTaskFromUI = () => {
         if (removeStrategy.includes(TASK_REMOVE_STRATEGY.STORAGE)) {
             removeTaskFromStorage(record)
         } else if (removeStrategy.includes(TASK_REMOVE_STRATEGY.PANEL)) {
             deleteTaskFromContext(record)
         }
+    }
+
+    const cancelTaskAndRemoveFromUI = storage.useUpdateTask({ status: TASK_STATUS.CANCELLED }, removeTaskFromUI)
+
+    const handleRemoveTask = useCallback(async () => {
+        // Tasks under processing should be cancelled before removing from UI
+        if (record.status === TASK_STATUS.PROCESSING) {
+            cancelTaskAndRemoveFromUI({}, record)
+        } else {
+            removeTaskFromUI()
+        }
     }, [record, storage, deleteTaskFromContext, removeTaskFromStorage])
```

Comment:
```suggestion
    }, [record, storage, removeTaskFromUI])
```

Reply:
`removeTaskFromUI` will not be changed, I does't needs to be presented in deps